File: ticket-frontend/src/modules/movie/dtos/movie-dto.ts
import { TypeUtils } from '../../core/type-utils';

export class MovieDto {
    movieId: number | null = null;
    name: string = '';
    description: string = '';
    image: string = '';
    imageName: string = '';
    imageMimePrefix: string = '';
    directors: string = '';
    script: string = '';
    actors: string = '';
    appropriateness: string = '';
    duration: number = 0;
    trailerSrcUrl: string = '';
    year: number = 0;

    static fromObj(obj: any): MovieDto | null {
        if (!obj) {
            return null;
        }
        const movieModel: MovieDto = new MovieDto();
        movieModel.movieId = obj.movieId;
        movieModel.name = obj.name;
        movieModel.description = obj.description;
        movieModel.image = obj.image;
        movieModel.imageName = obj.imageName;
        movieModel.imageMimePrefix = obj.imageMimePrefix;
        movieModel.directors = obj.directors;
        movieModel.script = obj.script;
        movieModel.actors = obj.actors;
        movieModel.appropriateness = obj.appropriateness;
        movieModel.duration = obj.duration;
        movieModel.trailerSrcUrl = obj.trailerSrcUrl;
        movieModel.year = obj.year;
        return movieModel;
    }

    static listFromObjList(objs: any[]): MovieDto[] {
        if (!objs) {
            return [];
        }
        return objs.map(MovieDto.fromObj).filter(TypeUtils.isNonNullable);
    }
}

File: ticket-backend/src/main/java/aics/domain/movie/dtos/MovieDto.java
package aics.domain.movie.dtos;

import aics.domain.movie.entities.Movie;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.Base64;

@Data
@Accessors(chain = true)
public class MovieDto implements Serializable {
    private Long movieId;
    private String name;
    private String description;
    private String image;
    private String imageName;
    private String imageMimePrefix;
    private String directors;
    private String script;
    private String actors;
    private String appropriateness;
    private int duration;
    private String trailerSrcUrl;
    private int year;

    public static MovieDto fromMovie(Movie movie) {
        if (movie == null) {
            return null;
        }
        return new MovieDto()
            .setMovieId(movie.getMovieId())
            .setName(movie.getName())
            .setDescription(movie.getDescription())
            .setImage(Base64.getEncoder().encodeToString(movie.getImage()))
            .setImageName(movie.getImageName())
            .setImageMimePrefix(movie.getImageMimePrefix())
            .setDirectors(movie.getDirectors())
            .setScript(movie.getScript())
            .setActors(movie.getActors())
            .setAppropriateness(movie.getAppropriateness())
            .setDuration(movie.getDuration())
            .setTrailerSrcUrl(movie.getTrailerSrcUrl())
            .setYear(movie.getYear());
    }
}
